solc/jsonCompiler.cpp

```diff
@@ -218,12 +218,13 @@ string compile(StringMap const& _sources, bool _optimize, CStyleReadFileCallback
 		{
 			// Do not taint the internal error list
 			ErrorList formalErrors;
-			if (compiler.prepareFormalAnalysis(&formalErrors))
+			ErrorReporter errorReporter(formalErrors);
+			if (compiler.prepareFormalAnalysis(&errorReporter))
 				output["formal"]["why3"] = compiler.formalTranslation();
-			if (!formalErrors.empty())
+			if (!errorReporter.errors().empty())
```

Comment:
The existence of `formalErrors` is a convenience I'd really like to do away with, actually.  Really, an `ErrorReporter` should be completely opaque and not even need to be passed an `ErrorList` in its constructor.

Ultimately, I want to go that route.  I guess you could call this code "in preparation" for that.  I could write this line in terms of `formalErrors`, but I'd rather avoid code that does much talking about `ErrorLists` where it's avoidable.